webrtc/interfaces.html

```diff
@@ -0,0 +1,74 @@
+<!doctype html>
+<html>
+  <head>
+    <title>"WebRTC IDL Tests</title>
+  </head>
+  <body>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/resources/WebIDLParser.js></script>
+    <script src=/resources/idlharness.js></script>
+    <script>
+      'use strict';
+
+      function initRTCCertificate(idlArray) {
+        return RTCPeerConnection.generateCertificate({
+          name: 'RSASSA-PKCS1-v1_5',
+          modulusLength: 2048,
+          publicExponent: new Uint8Array([1, 0, 1]),
+          hash: 'SHA-256'
+        })
+        .then(cert => {
+          window.rtcCert = cert;
+          idlArray.add_objects({'RTCCertificate': ['rtcCert']});
+         });
+      }
+
+      // A customized version of test_harness that returns
+      // promise of the test completing as well as allow
+      // other promise tests to run in parallel.
+      function promise_test(testFunc, testName, properties) {
+        const t = async_test(testName, properties);
+
+        return Promise.resolve()
+          .then(testFunc)
+          .then(
+            t.step_func_done(() => {}),
+            t.step_func(err => {
+              assert_unreached(`Unhandled rejection: ${err.toString()}`);
+            }))
+          .catch(() => {}); // ignore error since promise is always fulfilled
+      }
+
+      function doIdlTest(idlText) {
+        const idlArray = new IdlArray();
+
+        idlArray.add_untested_idls('interface EventHandler {};');
+        idlArray.add_untested_idls('interface MediaStreamTrack {};');
+        idlArray.add_idls(idlText);
+
+        // TODO: Add object for all IDL interfaces
+        idlArray.add_objects({
+          'RTCPeerConnection': ['new RTCPeerConnection'],
+          'RTCSessionDescription': ['new RTCSessionDescription({ type: "offer" })'],
+          'RTCIceCandidate': ['new RTCIceCandidate'],
+          'RTCPeerConnectionIceEvent': ['new RTCPeerConnectionIceEvent("ice")'],
+          'RTCPeerConnectionIceErrorEvent': ['new RTCPeerConnectionIceErrorEvent("ice-error", { errorCode: 701 });'],
+        });
+
+        return promise_test(
+          () => initRTCCertificate(idlArray),
+          'Test Driver for RTCCertificate')
+        .then(() => {
+          idlArray.test();
+        })
+      }
+
+      promise_test(() => {
+        return fetch('/interfaces/webrtc-pc.idl')
+          .then(response => response.text())
+          .then(doIdlTest);
```

Comment:
I think the structure is almost right now. If you make a `getRTCCertificate` return a promise that resolves with an RTCCertificate instance and doesn't do anything else, then you can do this, with the original `promise_test`:
```JavaScript
 promise_test(() => {
  return Promise.all([fetch('/interfaces/webrtc-pc.idl').then(response => response.text()),
                      getRTCCertificate()])
    .then(doIdlTest);
}, 'Test driver');
```

`doIdlTest` would take two arguments, `idlText` and `certificate` and use that to initialize the `idlArray` before finally calling `idlArray.test()`.